internal/service/advancedcluster/data_source_advanced_cluster.go

```diff
@@ -272,95 +271,43 @@ func dataSourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	clusterName := d.Get("name").(string)
 	useReplicationSpecPerShard := false
 	var replicationSpecs []map[string]any
-	var clusterID string
 
 	if v, ok := d.GetOk("use_replication_spec_per_shard"); ok {
 		useReplicationSpecPerShard = v.(bool)
 	}
 
-	if !useReplicationSpecPerShard {
-		clusterDescOld, resp, err := connV220240530.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
-		if err != nil {
-			if resp != nil {
-				if resp.StatusCode == http.StatusNotFound {
-					return nil
-				}
-				if admin20240530.IsErrorCode(err, "ASYMMETRIC_SHARD_UNSUPPORTED") {
-					return diag.FromErr(fmt.Errorf("please add `use_replication_spec_per_shard = true` to your data source configuration to enable asymmetric shard support. Refer to documentation for more details. %s", err))
-				}
-			}
-			return diag.FromErr(fmt.Errorf(errorRead, clusterName, err))
-		}
-
-		clusterID = clusterDescOld.GetId()
-
-		if err := d.Set("disk_size_gb", clusterDescOld.GetDiskSizeGB()); err != nil {
-			return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "disk_size_gb", clusterName, err))
-		}
-		clusterDescNew, _, err := connV2.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
-		if err != nil {
-			return diag.FromErr(fmt.Errorf(errorRead, clusterName, err))
-		}
-		if err := d.Set("replica_set_scaling_strategy", clusterDescNew.GetReplicaSetScalingStrategy()); err != nil {
-			return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "replica_set_scaling_strategy", clusterName, err))
-		}
-		if err := d.Set("redact_client_log_data", clusterDescNew.GetRedactClientLogData()); err != nil {
-			return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "redact_client_log_data", clusterName, err))
+	clusterDesc, resp, err := connV2.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			return nil
 		}
-
-		zoneNameToZoneIDs, err := getZoneIDsFromNewAPI(clusterDescNew)
-		if err != nil {
+		return diag.FromErr(fmt.Errorf(errorRead, clusterName, err))
+	}
+	zoneNameToOldReplicationSpecMeta, err := GetReplicationSpecAttributesFromOldAPI(ctx, projectID, clusterName, connV220240530.ClustersApi)
+	if err != nil {
+		if apiError, ok := admin20240530.AsError(err); !ok {
```

Comment:
instead of AsError, I think it's easier to use IsErrorCode, and keep in a var, e.g.:
```
isUnsupportedError := admin.IsErrorCode(err, "ASYMMETRIC_SHARD_UNSUPPORTED") 

if !isUnsupportedError ||  isUnsupportedError && useReplicationSpecPerShard) {
			return diag.FromErr(fmt.Errorf(errorRead, clusterName, err))
		} else if !useReplicationSpecPerShard {
			return diag.FromErr(fmt.Errorf("please add `use_replication_spec_per_shard = true` to your data source configuration to enable asymmetric shard support. Refer to documentation for more details. %s", err))
...

also if err != nil but these conditions are not met, shouldn't we send an error back?